pkg/receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -135,6 +134,14 @@ func (ghs *githubScraper) getCommitInfo(
 	var dels int = 0
 	var age int64 = 0
 
+	// We're using BehindBy here because we're comparing against the target
+	// branch, which is the default branch. In essence the response is saying
+	// the default branch is behind the queried branch by X commits which is
+	// the number of commits made to the queried branch but not merged into
+	// the default branch. Doing it this way involves less queries because
+	// we don't have to know the queried branch name ahead of time.
+	comPages := getNumPages(float64(100), float64(branch.Compare.BehindBy))
```

Comment:
Did we want to maintain the request to getNumPages here or bring up to speed with the rest of the function? See the [getBranches](https://github.com/liatrio/opentelemetry-collector-contrib/blob/gitprovider-impl-2/receiver/gitproviderreceiver/internal/scraper/githubscraper/helpers.go#L49-L69) function from the refactor.

Reply:
Here we need it because we have to know exactly how many commits back to go or else we would continue into the commits that existed from whatever it was branched from